src/main/java/com/exasol/validation/GitRepositoryValidator.java

```diff
@@ -4,35 +4,41 @@
 import java.util.*;
 import java.util.logging.Logger;
 
-import com.exasol.github.GitHubRepository;
+import com.exasol.GitRepository;
+import com.exasol.GitRepositoryContent;
 
 /**
- * This class checks if the GitHub project repository is ready for a release.
+ * Contains validations for a Git project.
  */
-public class GitHubProjectValidator implements ProjectValidator {
-    private static final Logger LOGGER = Logger.getLogger(GitHubProjectValidator.class.getName());
-    protected final GitHubRepository repository;
+public class GitRepositoryValidator {
+    private static final Logger LOGGER = Logger.getLogger(GitRepositoryValidator.class.getName());
+    private final GitRepository repository;
 
     /**
-     * Create a new instance of {@link GitHubProjectValidator}.
-     *
-     * @param repository repository to validate
+     * Create a new instance of {@link GitRepositoryValidator}.
+     * 
+     * @param repository instance of {@link GitRepository} to validate
      */
-    public GitHubProjectValidator(final GitHubRepository repository) {
+    public GitRepositoryValidator(final GitRepository repository) {
         this.repository = repository;
     }
 
-    @Override
-    public void validate() {
-        LOGGER.fine("Validating GitHub-specific requirements.");
-        final String changelog = this.repository.getChangelogFile();
-        final String version = this.repository.getVersion();
-        final String changes = this.repository.getChangesFile(version);
+    /**
+     * Validate content of a Git-based repository.
+     * 
+     * @param branch name of a branch to validate on
+     */
+    public void validate(final Optional<String> branch) {
+        LOGGER.fine("Validating Git repository.");
+        final GitRepositoryContent content = branch.isEmpty() ? this.repository.getRepositoryContent()
+                : this.repository.getRepositoryContent(branch.get());
+        final String changelog = content.getChangelogFile();
+        final String version = content.getVersion();
+        final String changes = content.getChangesFile(version);
         validateChangelog(changelog, version);
         validateChanges(changes, version);
-        final Optional<String> latestReleaseTag = this.repository.getLatestReleaseVersion();
+        final Optional<String> latestReleaseTag = this.repository.getLatestReleaseTag();
```

Comment:
A base Git repository has not concept of a release. It only knows tags.
Don't mix up GitHub features and Git features please. Otherwise you end up with tight coupling and broken SoC.

Reply:
It's only naming, but that is important too.